kdwsdl2cpp/common/fileprovider.cpp

```diff
@@ -94,30 +97,38 @@ bool FileProvider::get( const QUrl &url, QString &target )
     mFileName = target;
   }
 
-  qDebug("Downloading '%s'", url.toEncoded().constData());
-
-  QNetworkAccessManager manager;
-  QNetworkRequest request(url);
-  QNetworkReply* job = manager.get(request);
-
-  QEventLoop loop;
-  connect(job, SIGNAL(finished()),
-          &loop, SLOT(quit()));
-  loop.exec();
-
-  if (job->error()) {
-      qWarning("Error downloading '%s': %s", url.toEncoded().constData(), qPrintable(job->errorString()));
-      return false;
+  QByteArray data;
+  if (fileProviderCache.contains(url)) {
+    data = fileProviderCache[url];
+    qDebug( "Using cached copy" );
+  } else {
+    qDebug("Downloading '%s'", url.toEncoded().constData());
+
+    QNetworkAccessManager manager;
+    QNetworkRequest request(url);
+    QNetworkReply* job = manager.get(request);
+
+    QEventLoop loop;
+    connect(job, SIGNAL(finished()),
+            &loop, SLOT(quit()));
+    loop.exec();
+
+    if (job->error()) {
+        qWarning("Error downloading '%s': %s", url.toEncoded().constData(), qPrintable(job->errorString()));
+        return false;
+    }
+
+    qDebug( "Download successful" );
+    data = job->readAll();
+    fileProviderCache[url] = data;
```

Comment:
better written as `fileProviderCache.insert(url, data)`

I know, performance doesn't matter here, but it's just better practice. (reference: `Effective STL` book)

Reply:
This is a QMap, not an STL map. And since this is easier to read, and is not a performance matter, I'm leaving it.

Reply:
QMap has exactly the same behaviour as an STL map since this is a C++ issue. [] has to create an empty value which is then replaced with the actual value.

Easier to read is a matter of habit. Every time I see [] being used the alarm bells ring in my head because of unwanted detach (when reading) or because of this issue (when writing). value() and insert() solve both these issues.

But yes, OK, I'll accept the commit with [] if you won't change it ;)